### PR TITLE
Fix misleading icon import documentation

### LINKED ISSUE #5094: Misleading icon import documentation
**Describe the bug**

The [import section of the icons documentation](https://vanillaframework.io/docs/patterns/icons#import) suggests that users can specifically include directional chevrons and the information icon by name:
```scss
@include vf-p-icon-chevron-down;
@include vf-p-icon-chevron-up;
@include vf-p-icon-information;
```
However, Vanilla is not creating any mixins for specific directional chevrons. Rather, we create a single mixin [vf-p-icon-chevron ](https://github.com/canonical/vanilla-framework/blob/4109529d291740aa752b4e24967ef0d056d157ae/scss/_patterns_icons.scss#L131) that declares classes for the directional chevrons. The information icon is included in the mixin `vf-p-icon-info`, not `vf-p-icon-information`. 

Attempting to include these as directed by our documentation will lead to a build error as the mixins cannot be found.

**To Reproduce**

1. Icons documentation imports section
3. Observe the instructions to include the above mixins
4. Verify that there is no mixin in the Vanilla codebase matching any of the above.

**Expected behavior**

The icons import documentation accurately reflects how to import the minimum set of SCSS required to show each icon.

**Screenshots**
image

**Desktop (please complete the following information):**

- OS: PopOS 22.04
- Browser: Firefox 
- Version: 125.0.2

**Additional Context**
The icons documentation is creating the import statements by looping through `standard_icons` in `templates/docs/patterns/icons/index.md` (which also controls which icons are displayed in the standard & dark theme icons sections). 

Removing the instructions to import these nonexistent mixins would require slight changes around [here](https://github.com/canonical/vanilla-framework/blob/4109529d291740aa752b4e24967ef0d056d157ae/templates/docs/patterns/icons/index.md?plain=1#L134) to avoid creating import statements for specific mixins. Maybe it's easier to implement the mixins themselves?

Maybe we should create mixins for each direction, or remove the specific directional chevrons from the documentation and direct users to just include `vf-p-icon-chevron`.

### PR DESCRIPTION
**Pull request by jmuzina**
## Done
Fixes misleading documentation for importing icon mixins. The [current documentation](https://vanillaframework.io/docs/patterns/icons#import) directs users to import 3 mixins that do not exist (and would cause build errors if used):

- `vf-p-icon-chevron-up`
- `vf-p-icon-chevron-down`
- `vf-p-icon-information`

This updates the documentation to direct users to import `vf-p-icon-chevron` and `vf-p-icon-info` instead.
  
Fixes #5094, [WD-11768](https://warthogs.atlassian.net/browse/WD-11768)

### Problem
The [icons page](https://vanillaframework.io/docs/patterns/icons) uses the same [array of icon names](https://github.com/canonical/vanilla-framework/blob/d0197b27f5733789d6d23536eabe68356e045ae1/templates/docs/patterns/icons/index.md?plain=1#L21) to do two different things:

- Generate the icons sections that render the example icons to the page. These are rendered using a macro that uses the values from the array to [**create the class name**](https://github.com/canonical/vanilla-framework/blob/d0197b27f5733789d6d23536eabe68356e045ae1/templates/docs/patterns/icons/index.md?plain=1#L14) for the icon by prepending `p-icon--`.
- Generate the import instructions. Jinja loops over the elements in the icons array and [recreates mixin names](https://github.com/canonical/vanilla-framework/blob/d0197b27f5733789d6d23536eabe68356e045ae1/templates/docs/patterns/icons/index.md?plain=1#L134) by prepending `vf-p-icon--`.

The code is making the assumption that mixin names and icon class names are exactly the same except for the prefix `vf-` for mixins and `p-` for icons, and that each icon has its own dedicated mixin that matches this naming convention. This is mostly true, but is incorrect in three cases:
- `vf-p-icon-chevron-up` (mixin does not exist; class `p-icon--chevron-up` is generated by mixin `vf-p-icon-chevron`) ([src](https://github.com/canonical/vanilla-framework/blob/d0197b27f5733789d6d23536eabe68356e045ae1/scss/_patterns_icons.scss#L132))
- `vf-p-icon-chevron-down` (mixin does not exist; class `p-icon--chevron-down` is generated by mixin `vf-p-icon-chevron`) ([src](https://github.com/canonical/vanilla-framework/blob/d0197b27f5733789d6d23536eabe68356e045ae1/scss/_patterns_icons.scss#L136))
- `vf-p-icon-information` (mixin does not exist; class `p-icon--information` is generated by mixin `vf-p-icon-info`) ([src](https://github.com/canonical/vanilla-framework/blob/d0197b27f5733789d6d23536eabe68356e045ae1/scss/_patterns_icons.scss#L159))

#### Creating the missing mixins; deprecate the old ones (alternate / more permanent solution)
We could create mixins to match the import instructions. These would create duplicate CSS to the current mixins but would fix the incorrect documentation without editing the Jinja template. While **this PR does not do this**, we should probably change the mixin names at some point in the future (next major version perhaps) in the interest of consistency.

#### Altering the Jinja template to fix the imports (PR / temporary solution)
A much simpler fix that does not alter our SCSS would be to update the icons Jinja template to fix the import instructions. This makes the Jinja template a bit uglier but has no effect on our users' SCSS (besides pointing them towards the right mixins to import).

Note that my first approach for this solution was simply hard-coding the correct mixins. However this is a bit brittle for my taste so I added some logic to the standard imports loop in Jinja to solve the problem.

## QA

- Open [demo icons page](https://vanilla-framework-5100.demos.haus/docs/patterns/icons#import)
- Verify that imports for `vf-p-icon-chevron-up`, `vf-p-icon-chevron-down`, and `vf-p-icon-information` have been removed.
- Verify that imports for `vf-p-icon-chevron` and `vf-p-icon-info` have been added.
- Copy-paste the imports block into an IDE with SCSS support in a project that has Vanilla installed. Verify that all mixins are importable and your project builds.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots

Before:
Screenshot 2024-05-24 at 4 28 02 PM

After:
Screenshot 2024-05-24 at 4 27 51 PM

[WD-11768]: https://warthogs.atlassian.net/browse/WD-11768?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pastelcyborg — APPROVED** [review]

File: templates/docs/patterns/icons/index.md [hunk 1]
```diff
@@ -131,7 +131,14 @@ If you use a limited set of icons you may want to include them individually to r
 @include vf-p-icons-common;
 
 // include only the icons that you use in your project
-{% for icon in standard_icons + status_icons + social_icons %}@include vf-p-icon-{{ icon }};
+{% for icon in standard_icons + status_icons + social_icons if icon != 'chevron-down' -%}
```
Comment (bartaz) [inline templates/docs/patterns/icons/index.md:134]: As discussed in review before, would it be possible to move the logic from here somewhere to where lists of icons are created, and maybe create a separate list of `icon_mixins` to iterate through here?

Comment (jmuzina) [reply templates/docs/patterns/icons/index.md]: @bartaz 

Revised. Now I create a list of `base_icon_mixin_names` (contains the standard, status, & social icons). I didn't combine them all into a single `icon_mixins` list as the "additional icons" are visually separated from the rest in the import documentation and have a comment separately inserted, so I wanted to keep that in place.

**Review by pastelcyborg — APPROVED** [review]

**Comment by bartaz** [comment]
@jmuzina Would that need updating now that we have chevron-left and chevron-right merged?

**Comment by jmuzina** [comment]
> @jmuzina Would that need updating now that we have chevron-left and chevron-right merged?

Yes, I've just pushed an update to make it compatible with new chevron icons

File: templates/docs/patterns/icons/index.md [hunk 2]
```diff
@@ -26,6 +26,34 @@ context:
 
 {% set social_icons = ['facebook', 'instagram', 'twitter', 'linkedin', 'youtube', 'github', 'rss', 'email'] %}
 
+<!--
```
Comment (bartaz) [inline templates/docs/patterns/icons/index.md:29]: Let's use Jinja comments, so they don't end up in HTML

```suggestion
{#
```

File: templates/docs/patterns/icons/index.md [hunk 3]
```diff
@@ -26,6 +26,34 @@ context:
...
+    'chevron-up', 'chevron-down', 'chevron-left', 'chevron-right' are class names, not mixin names; instead use 'chevron' for mixin import
+    TODO We should fix this icon-mixin naming discrepancy in the next major version.
+    See https://github.com/canonical/vanilla-framework/pull/5100
+-->
```
Comment (bartaz) [inline templates/docs/patterns/icons/index.md:33]:
```suggestion
#}
```

File: templates/docs/patterns/icons/index.md [hunk 4]
```diff
@@ -26,6 +26,34 @@ context:
...
+
+{% set standard_icon_mixin_names = standard_icons
+  | map('replace', 'chevron-down', 'chevron')
+  | map('replace', 'chevron-up', 'chevron')
+  | map('replace', 'chevron-left', 'chevron')
+  | map('replace', 'chevron-right', 'chevron')
+  | list
+%}
+
+<!--
```
Comment (bartaz) [inline templates/docs/patterns/icons/index.md:43]:
```suggestion
{#
```

File: templates/docs/patterns/icons/index.md [hunk 5]
```diff
@@ -26,6 +26,34 @@ context:
...
+  'information' is a class name, not a mixin name; instead use 'info' for mixin import
+  TODO We should fix this icon-mixin naming discrepancy in the next major version.
+  See https://github.com/canonical/vanilla-framework/pull/5100
+-->
```
Comment (bartaz) [inline templates/docs/patterns/icons/index.md:47]:
```suggestion
#}
```

File: templates/docs/patterns/icons/index.md [hunk 6]
```diff
@@ -26,6 +26,34 @@ context:
...
+
+{% set status_icon_mixin_names = status_icons | map('replace', 'information', 'info') | list %}
+
+<!--
```
Comment (bartaz) [inline templates/docs/patterns/icons/index.md:51]:
```suggestion
{#
```

File: templates/docs/patterns/icons/index.md [hunk 7]
```diff
@@ -26,6 +26,34 @@ context:
...
+  Remove duplicates from the list of mixin names; otherwise there would be two 'chevron' entries
+-->
```
Comment (bartaz) [inline templates/docs/patterns/icons/index.md:53]:
```suggestion
#}
```

**Review by bartaz — COMMENTED** [review]
Small nit pick on using Jinja comments instead of HTML ones.

**Pull request merged**